src/main/scala/org/economicsl/auctions/sandbox.sc

```diff
@@ -53,4 +53,24 @@ val orderBook5 = orderBook4 + order8
 
 // take a look at paired orders
 val (pairedOrders, _) = orderBook5.takeWhileMatched
-pairedOrders.toList
\ No newline at end of file
+pairedOrders.toList
+
+
+// Why can this not be made covariant in T? Can divisible orders be constructed using an auxiliary constructor on LimitAskOrder?
+case class DivisibleLimitAskOrder[T <: Tradable](issuer: UUID, limit: Price, quantity: Quantity, tradable: T) extends
+  LimitAskOrder[T] with Divisible[T, DivisibleLimitAskOrder[T]] {
+
+  def split(residual: Quantity): (DivisibleLimitAskOrder[T], DivisibleLimitAskOrder[T]) = {
+    require(residual.value < this.quantity.value)  // can this be check be lifted into the type system?
```

Comment:
@davidrpugh Interesting. I like the idea of lifting the semantics of the check into the type system, that's a very "Haskellian" way of programming. It forces all functions dealing with the result of `split`to deal with the two cases. 

Regarding your second question: As far as I understand the Wellman paper, the split is always performed by the auction itself, right? If that is always the case, then it would probably be useful to make the method private, yes.